game/engine/main.py

```
@@ -183,13 +183,13 @@ def valid_pawn(color, fr, fc, tr, tc):
 
 
 def valid_rook(fr, fc, tr, tc):
-    return (fr == tr or fc == tc) and path_clear(fr, fc, tr, tc)
+    return fr == tr or fc == tc and path_clear(fr, fc, tr, tc)
```

Ruff (0.15.15)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)